Marlin/src/lcd/extensible_ui/ui_api.cpp

```diff
@@ -599,13 +599,15 @@ namespace ExtUI {
 
   void resumePrint() {
     #if ENABLED(SDSUPPORT)
+      ExtUI::onStatusChanged(PSTR(MSG_FILAMENT_CHANGE_RESUME_1));
```

Comment:
I would think a method named `onStatusChanged` would get called by the `MarlinUI` methods that change the status message, so that the extensible UI could pass the status message string on to the device that it implements, rather than being used like a `setStatus...` method. Perhaps it should be reworked in that manner.

And of course, since this is in the `ExtUI` namespace, the `ExtUI::` prefix is not needed here.

I will push a commit to fix that, plus one that simply calls `ui.reset_status` here and in `M24`, which is the appropriate method to get the status message to show the current filename or printing status.

Another note is that `MSG_FILAMENT_CHANGE_RESUME_1` is not obligated to be "Resuming" in every language. It might say "Print is now" as the first line of "Print is now resuming" since these strings are intended for full-screen modal displays. If a "`Resuming...`" status and some other standard status messages are desired, they would probably be better at the `MarlinUI` level, which would set the status message, and then —following the suggested handling of `onStatusChanged`— pass the updated status message on to `ExtUI`.

Reply:
Q: Can all our status message handlers be made to automatically differentiate between `char*` and progmem strings? Currently `PGM_P` is just a `typedef const char *`, but I see there is a `typedef const __FlashStringHelper *progmem_str` available, which can differentiate as long as the `F(...)` macro is used instead of `PSTR(...)`. If we can transition away from `PSTR(...)` and adopt `F(...)` that might neaten up a lot of code.

Reply:
@thinkyhead: I only used onStatusChanged directly because this is inside ExtUI and I figured it didn't matter. But the UltraLCD methods could be called as well, as they end up calling onStatusChanged indirectly.

I did come across one more weirdness with the resume/pause. If you "pause" the print with PARK_HEAD_ON_PAUSE enabled, it will move to the park position and retract the filament. If you resume, it will unretract and return to the print position, as it should. *HOWEVER*, if while the print is paused, you do an abort print. It unretracts, returns to the print position and then aborts. It would be nice if this was short-circuited in the case of an abort.

Reply:
Yeah, we should definitely patch up the abort behavior. And fix https://github.com/MarlinFirmware/Marlin/pull/12098#issuecomment-436231690 (aka #12411) while we're at it. I think the pause/resume/change features are a bit confusing because they're overloaded for multiple uses.

Reply:
> Q: Can all our status message handlers be made to automatically differentiate between `char*` and progmem strings? Currently `PGM_P` is just a `typedef const char *`

In the UI code I developed I tried to overload based on progmem strings. In some situations, it seems to have worked, in others it did not. I think the compiler doesn't overload on progmem per se, I think it overloads based on const-ness, which is not always what you want. Given that ambiguity, I don't recommend we go down that route with the rest of Marlin.

Reply:
Well the good thing about the flash-string-helper is that it encapsulates the strings in a class, and automatically returns a reference that normal string functions can utilize.